docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+export const meta = {
+  title: 'Component Architecture',
+}
+
+The design and architecture of components should follow from a set of first principles.
+
+- There is 1 right way to do things, and it's easy to do the right thing.
+- Consumers should not be penalized for code they don't use (through size, perf, etc).
+- Don't overgeneralize until we have a clear need (abstracting should solve a bundle size goal, or DRY goal, or partner use case)
+
+# Base components
+
+Each component should be built upon a "base component".
+
+In order to keep sorted lists formatted neatly, each base component should be named with the form: `ComponentNameBase.tsx`
```

Comment:
Let's exclude the idea of "Base Components" at all and let's try to generalize principles without any implementation details as it creates borders and edges for other proposals.

Reply:
Happy to discuss the pros/cons, but deciding on base component structure (including whether or not they even exist) is the main point of this doc. I think the borders and edges are the value proposition, as it makes theming/composition discussions clear and decoupled from component implementation discussions.

Reply:
Offline agreement: split this PR into arch requirements only (the "why"), then, as code is committed in separate PRs to fulfill those requirements, we can document "how" it is accomplished in Fluent UI.

This way, we're committing the description along with the real-world working thing.